src/nfa/nfa.cc

```diff
@@ -320,16 +320,47 @@ void Delta::add(State state_from, Symbol symbol, State state_to) {
     } else {
         const auto symbol_transitions{ state_transitions.find(Move{ symbol }) };
         if (symbol_transitions != state_transitions.end()) {
-            // Add transition with symbolOnTransition already used on transitions from stateFrom.
+            // Add transition with symbol already used on transitions from state_from.
             symbol_transitions->insert(state_to);
         } else {
-            // Add transition to a new Move struct with symbolOnTransition yet unused on transitions from stateFrom.
+            // Add transition to a new Move struct with symbol yet unused on transitions from state_from.
             const Move new_symbol_transitions{ symbol, state_to };
             state_transitions.insert(new_symbol_transitions);
         }
     }
 }
 
+void Delta::add(const State state_from, const Symbol symbol, const StateSet& states) {
```

Comment:
Is this already the best we can do or we ultimately need something better that avoids searching, like the same thing but parameterized with two iterators instead of state and symbol?

Reply:
We can add an overloaded function with two iterators. We will see whether we find usages for it in our algorithms.

Reply:
That was the question, maybe it is for vojta: could an iterator version save some searching?

Reply:
Rather no for the case of concatenation. But it might be useful later when we dive into optimization of other operations.